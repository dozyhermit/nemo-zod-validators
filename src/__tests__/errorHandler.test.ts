import {
  errorHandlerWithSchema,
  errorHandlerWithSchemaFormErrors,
} from '../errorHandler';

describe('errorHandlerWithSchema', () => {
  test('uses fieldErrors if in error', async () => {
    const result = errorHandlerWithSchema({
      // @ts-expect-error intentionally fudged, type errors expected
      error: {
        flatten: () => ({
          fieldErrors: { field: ['Field error exists!'] },
          formErrors: [],
        }),
      },
    });

    expect(await result.json()).toMatchSnapshot();
  });

  test('uses DEFAULT_ERROR_MESSAGE if flatten not in error', async () => {
    const result = errorHandlerWithSchema({
      // @ts-expect-error intentionally fudged, type errors expected
      error: {},
    });

    expect(await result.json()).toMatchSnapshot();
  });

  test('uses DEFAULT_ERROR_MESSAGE if fieldErrors not in flatten', async () => {
    const result = errorHandlerWithSchema({
      error: {
        // @ts-expect-error intentionally fudged, type errors expected
        flatten: () => ({}),
      },
    });

    expect(await result.json()).toMatchSnapshot();
  });

  test('uses DEFAULT_ERROR_MESSAGE if fieldErrors is in flatten but undefined', async () => {
    const result = errorHandlerWithSchema({
      error: {
        // @ts-expect-error intentionally fudged, type errors expected
        flatten: () => ({ fieldErrors: undefined }),
      },
    });

    expect(await result.json()).toMatchSnapshot();
  });

  test('uses DEFAULT_ERROR_MESSAGE if fieldErrors is in flatten but empty string', async () => {
    const result = errorHandlerWithSchema({
      error: {
        // @ts-expect-error intentionally fudged, type errors expected
        flatten: () => ({ fieldErrors: '' }),
      },
    });

    expect(await result.json()).toMatchSnapshot();
  });

  test('uses DEFAULT_ERROR_MESSAGE if fieldErrors is in flatten but empty object', async () => {
    const result = errorHandlerWithSchema({
      error: {
        // @ts-expect-error intentionally fudged, type errors expected
        flatten: () => ({ fieldErrors: {} }),
      },
    });

    expect(await result.json()).toMatchSnapshot();
  });
});

describe('errorHandlerWithSchemaFormErrors', () => {
  test('uses formErrors if in error', async () => {
    const result = errorHandlerWithSchemaFormErrors({
      error: {
        // @ts-expect-error intentionally fudged, type errors expected
        flatten: () => ({
          formErrors: ['Form error exists!'],
        }),
      },
    });

    expect(await result.json()).toMatchSnapshot();
  });

  test('uses DEFAULT_ERROR_MESSAGE if flatten not in error', async () => {
    const result = errorHandlerWithSchemaFormErrors({
      // @ts-expect-error intentionally fudged, type errors expected
      error: {},
    });

    expect(await result.json()).toMatchSnapshot();
  });

  test('uses DEFAULT_ERROR_MESSAGE if formErrors not in flatten', async () => {
    const result = errorHandlerWithSchemaFormErrors({
      error: {
        // @ts-expect-error intentionally fudged, type errors expected
        flatten: () => ({}),
      },
    });

    expect(await result.json()).toMatchSnapshot();
  });

  test('uses DEFAULT_ERROR_MESSAGE if formErrors is in flatten but undefined', async () => {
    const result = errorHandlerWithSchemaFormErrors({
      error: {
        // @ts-expect-error intentionally fudged, type errors expected
        flatten: () => ({ formErrors: undefined }),
      },
    });

    expect(await result.json()).toMatchSnapshot();
  });

  test('uses DEFAULT_ERROR_MESSAGE if formErrors is in flatten but empty string', async () => {
    const result = errorHandlerWithSchemaFormErrors({
      error: {
        // @ts-expect-error intentionally fudged, type errors expected
        flatten: () => ({ formErrors: '' }),
      },
    });

    expect(await result.json()).toMatchSnapshot();
  });

  test('uses DEFAULT_ERROR_MESSAGE if formErrors is in flatten but empty object', async () => {
    const result = errorHandlerWithSchemaFormErrors({
      error: {
        // @ts-expect-error intentionally fudged, type errors expected
        flatten: () => ({ formErrors: {} }),
      },
    });

    expect(await result.json()).toMatchSnapshot();
  });
});
