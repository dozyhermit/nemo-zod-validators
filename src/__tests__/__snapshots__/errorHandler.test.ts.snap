// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`errorHandlerWithSchema uses DEFAULT_ERROR_MESSAGE if fieldErrors is in flatten but empty object 1`] = `
{
  "message": "Validation has failed",
}
`;

exports[`errorHandlerWithSchema uses DEFAULT_ERROR_MESSAGE if fieldErrors is in flatten but empty string 1`] = `
{
  "message": "Validation has failed",
}
`;

exports[`errorHandlerWithSchema uses DEFAULT_ERROR_MESSAGE if fieldErrors is in flatten but undefined 1`] = `
{
  "message": "Validation has failed",
}
`;

exports[`errorHandlerWithSchema uses DEFAULT_ERROR_MESSAGE if fieldErrors not in flatten 1`] = `
{
  "message": "Validation has failed",
}
`;

exports[`errorHandlerWithSchema uses DEFAULT_ERROR_MESSAGE if flatten not in error 1`] = `
{
  "message": "Validation has failed",
}
`;

exports[`errorHandlerWithSchema uses fieldErrors if in error 1`] = `
{
  "message": {
    "field": [
      "Field error exists!",
    ],
  },
}
`;

exports[`errorHandlerWithSchemaFormErrors uses DEFAULT_ERROR_MESSAGE if flatten not in error 1`] = `
{
  "message": "Validation has failed",
}
`;

exports[`errorHandlerWithSchemaFormErrors uses DEFAULT_ERROR_MESSAGE if formErrors is in flatten but empty object 1`] = `
{
  "message": "Validation has failed",
}
`;

exports[`errorHandlerWithSchemaFormErrors uses DEFAULT_ERROR_MESSAGE if formErrors is in flatten but empty string 1`] = `
{
  "message": "Validation has failed",
}
`;

exports[`errorHandlerWithSchemaFormErrors uses DEFAULT_ERROR_MESSAGE if formErrors is in flatten but undefined 1`] = `
{
  "message": "Validation has failed",
}
`;

exports[`errorHandlerWithSchemaFormErrors uses DEFAULT_ERROR_MESSAGE if formErrors not in flatten 1`] = `
{
  "message": "Validation has failed",
}
`;

exports[`errorHandlerWithSchemaFormErrors uses formErrors if in error 1`] = `
{
  "message": [
    "Form error exists!",
  ],
}
`;
